app/src/main/java/org/p2p/wallet/auth/ui/restore/common/CommonRestorePresenter.kt

```diff
@@ -25,11 +28,31 @@ class CommonRestorePresenter(
     override fun setGoogleIdToken(userId: String, idToken: String) {
         view?.setLoadingState(isScreenLoading = true)
         socialShareRestoreInteractor.restoreSocialShare(idToken, userId)
-        view?.setLoadingState(isScreenLoading = false)
         if (restoreUserRestoreInteractor.isUserReadyToBeRestored()) {
-            // navigate to pin if social + device share exists
+            launch {
+                restoreUserWithShares()
+            }
         } else {
             view?.navigateToPhoneEnter()
+            view?.setLoadingState(isScreenLoading = false)
+        }
+    }
+
+    private suspend fun restoreUserWithShares() {
+        val result = restoreUserRestoreInteractor.tryRestoreUser(
+            UserRestoreInteractor.RestoreUserWay.DevicePlusSocialShareWay
+        )
+        when (result) {
+            is UserRestoreInteractor.RestoreUserResult.RestoreSuccessful -> {
+                restoreUserRestoreInteractor.finishAuthFlow()
+                view?.setLoadingState(isScreenLoading = false)
+                view?.navigateToPinCreate()
+            }
+            is UserRestoreInteractor.RestoreUserResult.RestoreFailed -> {
+                Timber.e(result)
+                view?.setLoadingState(isScreenLoading = false)
+                view?.showUiKitSnackBar(messageResId = R.string.error_general_message)
```

Comment:
if shares not match - we show alert dialog "Shares not match"

Reply:
not needed on android side